### ISSUE TITLE
Tabbing skips the blocks UI in various workflows

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#11773]
This is something difficult to describe so I've made a GIF. Seems to me it doesn't work as expected starting with 4.2.0: I've tested in 4.1.0 and everything works as expected.

Steps followed in the the GIF below:
- press the Tab key to navigate through the UI
- as expected, tabbing navigates through the whole blocks UI including the movers and the toolbar buttons
- then, press the navigate region shortcut `Ctrl + backtick` twice to jump to the content area
- press the Tab key again to navigate through the UI
- this time, the blocks UI gets skipped and only the editable fields get the focus

I haven't been able to identify the root cause.

![tabbing](https://user-images.githubusercontent.com/1682452/48368776-ebd66c00-e6b4-11e8-94d9-ac96a3213392.gif)

**Comment by afercia** [comment]
I've just realized this happens also in other flows, not only when using the navigate regions shortcut.

In the GIF below, I'm using the same testing tool used previously to highlight the focused element with a red outline. Using a simple post with paragraphs and one heading:
- the first time, tabbing through the UI works as expected: all the block UI elements are navigable with a keyboard and receive focus
- then I just click outside of the blocks list, in an empty area on the left
- press the Tab key again
- this time, the blocks UI gets skipped and only the editable fields get the focus (except for the last block)

![tabbing skips ui](https://user-images.githubusercontent.com/1682452/49002959-3e0d9780-f161-11e8-994b-787c7966c2ff.gif)

This appears to be a serious regression as it makes impossible to properly use Gutenberg with only the keyboard. There are probably more scenarios where this happens.

@mtias I'm afraid this regression can't be milestoned for "WordPress 5.0.x Follow Ups" as right now the UI is unusable with the keyboard in unexpected ways. I'd strongly recommend to give this issue a higher priority as releasing Gutenberg with a broken keyboard navigation is far from ideal.

**Comment by afercia** [comment]
Updated the title to better reflect the issue.
